### PR TITLE
assertion failed : nWitnessCasheSize > 0,

### PR DESCRIPTION
**Pull request by ezzygarmyz**

### CODE REVIEW

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -1242,13 +1250,19 @@ void DecrementNoteWitnesses(NoteDataMap& noteDataMap, int indexHeight, int64_t n
 void CWallet::DecrementNoteWitnesses(const CBlockIndex* pindex)
 {
     LOCK(cs_wallet);
-    for (std::pair<const uint256, CWalletTx>& wtxItem : mapWallet) {
+    bool hasSprout = false;
+    bool hasSapling = false;
+    or (std::pair<const uint256, CWalletTx>& wtxItem : mapWallet) {
```
Comment (MarkLTZ) [inline src/wallet/wallet.cpp]: Syntax error.

Comment (ezzygarmyz) [reply src/wallet/wallet.cpp]: yes right, its 'for' not 'or'

Comment (nucash-mining) [reply src/wallet/wallet.cpp]: heh

Comment (nucash-mining) [reply src/wallet/wallet.cpp]: heh

**Pull request closed without merging**